<?xml version="1.0"?>

<launch>
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find final_assignment)/param/costmap_common_params.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find final_assignment)/param/costmap_common_params.yaml" command="load" ns="local_costmap"/>
    <rosparam file="$(find final_assignment)/param/local_costmap_params.yaml" command="load"/>
    <rosparam file="$(find final_assignment)/param/global_costmap_params.yaml" command="load"/>
    <rosparam file="$(find final_assignment)/param/base_local_planner_params.yaml" command="load"/>
    <rosparam file="$(find final_assignment)/param/move_base_params.yaml" command="load"/>
     
    <arg name="des_x" default= "1"/>
    <arg name="des_y" default= "1"/>
    <param name="des_pos_x" value="$(arg des_x)" />
    <param name="des_pos_y" value="$(arg des_y)" />
    <node pkg="gmapping" type="my_tf_echo" name="tf_echo" />


    <node pkg="final_assignment" type="wall_follow_service_m.py" name="wall_follower" output="screen"/>
    <node pkg="final_assignment" type="robot.py" name="robot" output="screen"/>
    <node pkg="final_assignment" type="user_interface.py" name="user_interface" output="screen" />
    <node pkg="final_assignment" type="random.py" name="random" output="screen"/>

  </node>
</launch>
